### PR TITLE
Riande mvc mig1

### PR DESCRIPTION
**Pull request by Rick-Anderson**
@danroth27 @ardalis @Erikre @rachelappel  review

### CODE REVIEW

File: mvc/migration/migratingfrommvc5.rst [hunk 1]
```diff
@@ -1,257 +1,244 @@
 Migrating From ASP.NET MVC 5 to MVC 6
 =====================================
 
-By `Steve Smith`_ 
+By `Rick Anderson`_ and `Steve Smith`_
 
-Migrating from ASP.NET MVC 5 to ASP.NET 5 and MVC 6 requires a few steps to complete, since ASP.NET 5 introduces a number of new concepts. In this article you will learn how to migrate from the ASP.NET MVC 5 default project template to ASP.NET MVC 6, including initial setup, basic controllers and views, static content, and client side dependencies.
+While this article shows how to migrate an ASP.NET MVC 5 project to ASP.NET MVC 6, it's arguably more valuable in showing what has changed from MVC 5 to MVC 6. The migration process in this (part 1 tutorial) includes initial setup, basic controllers and views, static content, and client side dependencies. The tutorial creates the MVC 6 app with the empty template and adds all the assets from the legacy MVC 5 app needed to complete the conversion. An alternative approach to migrations is to create the MVC 6 app with the full MVC app template, and then copy the assets and your custom code.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Clarify where to "copy the assets and your custom code".

File: mvc/migration/migratingfrommvc5.rst [hunk 2]
```diff
@@ -1,257 +1,244 @@
...
 
-In this article:
-	- `Create the Initial Project`_
-	- `Create the Destination Solution`_
-	- `Migrate Basic Controllers, Views, and Static Content`_
-	- `Configure Bundling`_
+	- `Create the starter MVC 5 project`_
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: This list should have an intro sentence. If any editor says differently, let me talk to them.

File: mvc/migration/migratingfrommvc5.rst [hunk 3]
```diff
@@ -1,257 +1,244 @@
...
+	- `Create the MVC 6  project`_
+	- `Configure the site to use MVC`_
+	- `Controllers and views`_
+	- `Static content`_
+	- `Migrate the layout file`_
 
-Download the `finished source <https://github.com/aspnet/Docs/tree/master/samples/MigratingFromMvc5/NewMvc6Project>`_ from the project created in this article.
+Create the starter MVC 5 project
+========================================
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Looks like the length of this section underline doesn't match the length of the section title. I thought that was a requirement of restructured text.

Comment (Rick-Anderson) [inline mvc/migration/migratingfrommvc5.rst]: >=

File: mvc/migration/migratingfrommvc5.rst [hunk 4]
```diff
@@ -1,257 +1,244 @@
...
 
-Create the Initial Project
---------------------------
-
-For the purposes of this article, we will be starting from the default ASP.NET MVC 5 starter web project, which you can create in Visual Studio 2015 by adding a new web project and choosing MVC 5.
+To demonstrate the upgrade, we'll start by creating an new ASP.NET MVC 5 app. Create it with the name *WebApp1* so the namespace will match the MVC 6 project we create in the next step.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Should not be using "we'll", you should use "you'll". Same goes with "we" -> "you".

File: mvc/migration/migratingfrommvc5.rst [hunk 5]
```diff
@@ -1,257 +1,244 @@
...
 .. image:: migratingfrommvc5/_static/new-project-select-mvc-template.png
 
-If you prefer, you can `view or download the MVC 5 Project used in this article <https://github.com/aspnet/Docs/tree/master/samples/MigratingFromMvc5/Mvc5Project>`_.
-
-This sample web project will demonstrate how to migrate an MVC 5 web project that includes controllers, views, and ASP.NET Identity models, as well as startup and configuration logic common to many MVC 5 projects.
+The completed sample (this tutorial and the next two) will demonstrate how to migrate an MVC 5 web project that includes controllers, views, ASP.NET Identity models, startup and configuration code found in many MVC 5 projects.
 
-Create the Destination Solution
--------------------------------
+*Optional:* Change the name of the Solution from *WebApp1* to *Mvc5*. Visual Studio will display the new solution name (*Mvc5*), which will make it easier to tell this project from the next project.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: This should be formatted the same way a note is formatted.

File: mvc/migration/migratingfrommvc5.rst [hunk 6]
```diff
@@ -1,257 +1,244 @@
...
-Create the Destination Solution
--------------------------------
+*Optional:* Change the name of the Solution from *WebApp1* to *Mvc5*. Visual Studio will display the new solution name (*Mvc5*), which will make it easier to tell this project from the next project. 
 
-We will begin our migration by creating a new, empty ASP.NET 5 solution. Create a new project in Visual Studio 2015, choose an ASP.NET Web Application, and then choose the ASP.NET 5 Empty template.
+Create the MVC 6 project
+========================================
 
-.. image:: migratingfrommvc5/_static/new-project-mvc6.png
+Create a new empty MVC 6 web app with the same name as the previous project (*WebApp1*) so the namespaces in the two projects match. Having the same namespace makes it easier to copy code between the two projects. You'll have to create this project in a different directory than the previous project to use the same name.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Looks like *WebApp1* has a formatting issue. This should be fixed.

File: mvc/migration/migratingfrommvc5.rst [hunk 7]
```diff
@@ -1,257 +1,244 @@
...
 
-This migration will start from an empty template. If you're already familiar with ASP.NET 5 and its starter templates and there are features in a starter template you would like to take advantage of, you may wish to start from another template. The next step is to configure the site to use MVC. This requires changes to the project.json file and Startup.cs file. First, open project.json and add "Microsoft.AspNet.Mvc" to the "dependencies" property:
+- *Optional:* Create a new MVC 6 app named *WebApp1* with the full template. You can look at the template generated code to see the end result or to copy code to the conversion project. It's also helpful when you get stuck on a conversion step to compare with the template generated project.
+
+Configure the site to use MVC
+========================================
+
+- Open the *project.json* file and add ``Microsoft.AspNet.Mvc`` to the ``dependencies`` property and the scripts section as highlighted below:
+
+.. literalinclude:: migratingfrommvc5/samples/WebApp1/src/WebApp1/project.json
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: This include is missing.

File: mvc/migration/migratingfrommvc5.rst [hunk 8]
```diff
@@ -1,257 +1,244 @@
...
+	:language: json
+	:emphasize-lines: 8-9, 29-31
+  
+``Microsoft.AspNet.StaticFiles`` is the static file handler. The ASP.NET runtime is modular, and you must explicitly opt in to serve static files.
 
-.. code-block:: javascript
+The ``scripts`` section is used to specify when specified build automation scripts should run. Visual Studio now has built-in support for running scripts before and after specific events. The ``scripts`` section above specifies `NPM <https://docs.npmjs.com/>`__, `Bower <http://bower.io/>`__ and `Gulp <http://gulpjs.com/>`__ scripts should run on the ``prepublish`` stage.  We'll talk about NPM, Bower, and Gulp later in the tutorial.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Seems you should explain more about prepublish or point to more information about how this is done or the available options. Tough to see without the above json.

File: mvc/migration/migratingfrommvc5.rst [hunk 9]
```diff
@@ -1,257 +1,244 @@
...
-		"Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
-		"Microsoft.AspNet.Mvc": "6.0.0-beta4"
-	},
+For more information see `project.json <https://docs.asp.net/en/latest/conceptual-overview/understanding-aspnet5-apps.html#the-project-json-file>`_ and `Introducing .NET Core <http://docs.asp.net/en/latest/conceptual-overview/dotnetcore.html>`_.
 
-Now open Startup.cs and modify it as follows:
+- Open the *Startup.cs* file and change the code to match the following:
 
-.. code-block:: c#
+.. literalinclude:: migratingfrommvc5/samples/WebApp1/src/WebApp1/Startup.cs
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: This include does not show up!

File: mvc/migration/migratingfrommvc5.rst [hunk 10]
```diff
@@ -1,257 +1,244 @@
...
-				template: "{controller=Home}/{action=Index}/{id?}");
-		});
-	}
+``UseStaticFiles`` adds the static file handlers. As mentioned previously, the ASP.NET runtime is modular, and you must explicitly opt in to serve static files. For more information, see `Application Startup <http://docs.asp.net/en/latest/fundamentals/startup.html>`_ and `Routing <https://docs.asp.net/en/latest/fundamentals/routing.html>`__.
 
-At this point we are ready to create a simple Controller and View. Add a Controllers folder and a Views folder to the project. Add an MVC Controller called HomeController.cs class to the Controllers folder and a Home folder in the Views folder. Finally, add an Index.cshtml MVC View Page to the Views/Home folder. The project structure should be as shown:
+Add a controller and view
+========================================
+
+In this section you'll add a minimal controller and view to serve as place holders for the MVC 5 controller and views you'll migrate in the next section.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: "In this section" follows the same pattern as "In this topic" or "in this tutorial". Why is one okay and the other not?

File: mvc/migration/migratingfrommvc5.rst [hunk 11]
```diff
@@ -1,257 +1,244 @@
...
-Run the application - you should see Hello World output in your browser.
+Run the app.
 
 .. image:: migratingfrommvc5/_static/hello-world.png
 
 .. _migrate-controllers-views-static:
 
-Migrate Basic Controllers, Views, and Static Content
-----------------------------------------------------
+See :doc:`../controllers/index` and :doc:`../views/index` for more information.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: The link text is odd. Did you mean for these links to be this way?

File: mvc/migration/migratingfrommvc5.rst [hunk 12]
```diff
@@ -1,257 +1,244 @@
...
+
+Now that we have a minimal working MVC 6 project, we can start migrating functionality from the MVC 5 project. We will need to move the following:
+
+- client-side content (CSS, fonts, and scripts)
+- controllers
+- views
+- models
+- bundling 
+- filters
+- Log in/out, identity (This will be done in the next tutorial.)
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Shouldn't you capitalize the first character for each line in this list. Do did that on the last list.

File: mvc/migration/migratingfrommvc5.rst [hunk 13]
```diff
@@ -1,257 +1,244 @@
...
+
+Controllers and views
+========================================
+
+- Copy each of the methods from the MVC 5 ``HomeController`` to the MVC 6 ``HomeController``. Note that in MVC 5, the built in template generated return type is `ActionResult <https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ActionResult.cs>`_, in MVC 6 the templates generate `IActionResult <https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Abstractions/IActionResult.cs>`__ methods. ``ActionResult`` is the only implementation of ``IActionResult``, so there is no need to change the return type of your action methods.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: You are starting the "Controllers and views" section without an intro sentence. Add a sentence to tell the reader what you are explaining here.

File: mvc/migration/migratingfrommvc5.rst [hunk 14]
```diff
@@ -1,257 +1,244 @@
...
+- Delete the *Views/Home/Index.cshtml* view in the MVC 6 project.
+- Copy the *About.cshtml*, *Contact.cshtml* and *Index.cshtml* razor view files from the MVC 5 project to the MVC 6 project. 
+- Run the MVC 6 app and test each method. We haven't migrated the layout file or styles yet, so the rendered views will only contain the content in the view files. You won't have the layout file generated links for the ``About`` and ``Contact`` views, so you'll have to invoke them from the browser (replace **2468** with the port number used in your project).
+- 
+	- http://localhost:2468/home/about
+	- http://localhost:2468/home/contact
+	
+.. image:: migratingfrommvc5/_static/contact-page.png
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Is the image part of the list? If so, it should be added to the list.

File: mvc/migration/migratingfrommvc5.rst [hunk 15]
```diff
@@ -1,257 +1,244 @@
...
 
-Now that we've confirmed we have a simple, working ASP.NET MVC 6 project, it's time to start migrating functionality from the source project. There are many different ways one can approach this task. We will need to move all of the client-side content files (CSS, fonts, scripts), all of the controllers, views, and models, and migrate configured features like bundling, filters, and identity. Let's begin by replacing our simple "hello world" implementation of HomeController with the actual HomeController and Views from the source project.
+Note the lack of styling and menu items, we'll fix that in the next section.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Format as a note.

File: mvc/migration/migratingfrommvc5.rst [hunk 16]
```diff
@@ -1,257 +1,244 @@
...
+Static content
+========================================
 
-Next, create new MVC View Pages in the Views/Home folder for About and Contact. Copy the contents of the corresponding views in the old project to these new views, as well as Index.cshtml. At this point you should once again be able to run the new application, and although the styles are not yet in place, you should see the correct content on the home page as well as /home/about and /home/contact (contact is shown here):
+In previous versions of MVC (including MVC 5), static content was hosted from the root of the web project and was intermixed with server-side files. In MVC 6, static content is hosted in the *wwwroot* folder. You'll want to copy the static content from your MVC 5 app to the *wwwroot* folder in your MVC 6 project. In this sample conversion:
 
-.. image:: migratingfrommvc5/_static/contact-page.png
+- Copy the *favicon.ico* file the MVC 5 project to the *wwwroot* folder in the MVC 6 project.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Looks like you're missing the word "in" in this bullet item.

File: mvc/migration/migratingfrommvc5.rst [hunk 17]
```diff
@@ -1,257 +1,244 @@
...
-
-.. literalinclude:: migratingfrommvc5/samples/NewMvc6Project/src/NewMvc6Project/package.json
-	:language: javascript
-	:emphasize-lines: 6-8
-	:linenos: 
+The MVC 5 project uses `Bootstrap <http://getbootstrap.com/>`__ for its styling, and stores the Bootstrap files in the *Content* and *Scripts* folders. The template generated MVC 5 project references Bootstrap in the layout file (*Views/Shared/_Layout.cshtml*). You could copy the *bootstrap.js* and *bootstrap.css* files from the MVC 5 project to the *wwwroot* folder in the new project, but that approach doesn't use the improved mechanism for managing client-side dependencies in ASP.NET 5.
 
-Save your changes. You should see a new NPM folder in your project, under Dependencies, and it should now include gulp (3.8.11) as well as the related packages. In addition to Gulp itself, these two packages will allow us to clean up folders before we copy files to them, and to concatenate two or more files together, to achieve bundling.
+In the new project, we'll add support for Bootstrap (and other client-side libraries) using `Bower <http://bower.io/>`__ and `Grunt <http://gruntjs.com/>`__.
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Don't use "we" or "we'll". I think Mr. Pope said this to me 576 times.
Also, should you be using Gulp over Grunt?

File: mvc/migration/migratingfrommvc5.rst [hunk 18]
```diff
@@ -1,257 +1,244 @@
...
 
-Next, add a new Gulp Configuration file (Gulpfile.js) to the root of the project.
+- Add a `Bower <http://bower.io/>`__ configuration file named *bower.json* to the project root (Right click on the project, and then **Add > New Item > Bower Configuration File**).
```
Comment (Erikre) [inline mvc/migration/migratingfrommvc5.rst]: Don't put instructions in ( ). Instead:
Add a `Bower <http://bower.io/>`__ configuration file named *bower.json* to the project root by right-clicking on the project, and then **Add > New Item > Bower Configuration File**.

**Comment by danroth27** [comment]
@Rick-Anderson There are over 700 files in this PR, which is too big to review on GitHub. Can you break these changes up into several different PRs?

**Comment by Rick-Anderson** [comment]
@danroth27 another huge flaw with GitHub - How is that possible? I wouldn't know how to break it up, all the vital info is in the .rst. I can make word/html docs for anyone wanting to review it.

**Comment by danroth27** [comment]
Needs to be updated for RC1.

**Comment by Rick-Anderson** [comment]
submitted different pr

**Pull request closed without merging**